php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1657 @@
+      return new ArrayPairTreeImpl(expression, pairExpression.get().first(), pairExpression.get().second());
+    }
+    return new ArrayPairTreeImpl(expression);
+  }
+
+  public ArrayPairTree arrayPair2(ReferenceVariableTree referenceVariableTree) {
+    return new ArrayPairTreeImpl(referenceVariableTree);
+  }
+
+  public SeparatedList<ArrayPairTree> arrayInitializerList(
+    ArrayPairTree firstElement,
+    Optional<List<Tuple<InternalSyntaxToken, ArrayPairTree>>> restElements,
+    Optional<InternalSyntaxToken> trailingComma
+  ) {
+    return separatedList(firstElement, restElements, trailingComma.orNull());
+  }
+
+  public ArrayInitializerTree newArrayInitFunction(
+    InternalSyntaxToken arrayToken, InternalSyntaxToken openParenthesis,
+    Optional<SeparatedList<ArrayPairTree>> elements,
+    InternalSyntaxToken closeParenthesis
+  ) {
+    return new ArrayInitializerFunctionTreeImpl(
+      arrayToken,
+      openParenthesis,
+      elements.isPresent() ? elements.get() : new SeparatedList<>(ImmutableList.<ArrayPairTree>of(), ImmutableList.<InternalSyntaxToken>of()),
+      closeParenthesis);
+  }
+
+  public ArrayInitializerTree newArrayInitBracket(InternalSyntaxToken openBracket, Optional<SeparatedList<ArrayPairTree>> elements, InternalSyntaxToken closeBracket) {
+    return new ArrayInitializerBracketTreeImpl(
+      openBracket,
+      elements.isPresent() ? elements.get() : new SeparatedList<>(ImmutableList.<ArrayPairTree>of(), ImmutableList.<InternalSyntaxToken>of()),
+      closeBracket);
+  }
+
+  public FunctionExpressionTree functionExpression(Optional<InternalSyntaxToken> staticToken, InternalSyntaxToken functionToken, Optional<InternalSyntaxToken> ampersandToken,
+    ParameterListTree parameters, Optional<LexicalVariablesTree> lexicalVariables, BlockTree block) {
+
+    return new FunctionExpressionTreeImpl(
+      staticToken.orNull(),
+      functionToken,
+      ampersandToken.orNull(),
+      parameters,
+      lexicalVariables.orNull(),
+      block);
+  }
+
+  public NewExpressionTree newExpression(InternalSyntaxToken newToken, ExpressionTree expression) {
+    return new NewExpressionTreeImpl(newToken, expression);
+  }
+
+  public ExitTreeImpl newExitExpression(InternalSyntaxToken openParenthesis, Optional<ExpressionTree> expressionTreeOptional, InternalSyntaxToken closeParenthesis) {
+    return new ExitTreeImpl(openParenthesis, expressionTreeOptional.orNull(), closeParenthesis);
+  }
+
+  public ExitTree completeExitExpression(InternalSyntaxToken exitOrDie, Optional<ExitTreeImpl> partial) {
+    return partial.isPresent() ? partial.get().complete(exitOrDie) : new ExitTreeImpl(exitOrDie);
+  }
+
+  public ExpressionTree combinedScalarOffset(ArrayInitializerTree arrayInitialiser, Optional<List<ArrayAccessTree>> offsets) {
+    ExpressionTree result = arrayInitialiser;
+    for (ArrayAccessTree offset : optionalList(offsets)) {
+     result = ((ArrayAccessTreeImpl) offset).complete(result);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 7. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)